src/clapsolver/dynamic_lsap.py

```diff
@@ -0,0 +1,261 @@
+import numpy as np
+
+
+def augment(cost_matrix, curRow, row4col, col4row, u, v):
+    """Find the shortest augmenting path for the selected row. Update the dual
+       variables. Augment the previous solution.
+
+    Parameters
+    ----------
+    cost_matrix : 2darray
+         A matrix of costs.
```

Comment:
Specify any constraints on the matrix. Are rectangular matrices okay? Any shape? What about `np.inf` or `np.nan` entries? Are those allowed?